components/main_page/commons/main_page_layout.lua

```diff
@@ -68,11 +68,12 @@ function MainPageLayout._makeCells(cells)
 		for _, item in ipairs(column.children) do
 			local content = {}
 			if item.content then
+				local contentBody = item.content.body
 				if item.content.noPanel then
-					table.insert(content, frame:preprocess(item.content.body))
+					table.insert(content, type(contentBody) == 'string' and frame:preprocess(contentBody) or tostring(contentBody))
 				else
 					table.insert(content, tostring(PanelWidget{
-						body = frame:preprocess(item.content.body),
+						body = type(contentBody) == 'string' and frame:preprocess(contentBody) or contentBody,
```

Comment:
why no tostring here while you use it a few lines above?

mind when we make the grid stuff widget we can simplify stuff

possibly extract to function since basically is the same as a few lines above